boxsdk/object/base_endpoint.py

```diff
@@ -1,18 +1,21 @@
 # coding: utf-8
 
-from __future__ import unicode_literals
+from __future__ import unicode_literals, absolute_import
 
 
 class BaseEndpoint(object):
     """A Box API endpoint."""
 
-    def __init__(self, session):
+    def __init__(self, session, response_object=None):
         """
-
         :param session:
             The Box session used to make requests.
         :type session:
             :class:`BoxSession`
+        :param response_object:
+            The Box API response representing the object.
+        :type response_object:
+            :class:`BoxResponse`
         """
         self._session = session
```

Comment:
Though, I would recommend adding back

```python
super(BaseEndpoint, self).__init__(**kwargs)
```

since we're now dealing with multiple-inheritance.

Add ``**kwargs`` to the method declaration. You might possibly that in some of the other ``__init__`` methods too.